### PR TITLE
Add Dictionary to Env when Creating dictionary

### PR DESCRIPTION
**Pull request by ankurtrivedi**

### CODE REVIEW

File: src/main/resources/xlr_xldeploy/XLDeployClient.py [hunk 1]
```diff
@@ -373,15 +373,21 @@ def update_ci_property(self, ci_id, ci_property, property_value):
...
         ci = self.get_ci(env_id, 'json')
         data = json.loads(ci)
-        data["members"].append(ci_id)
+        if str(ci_type) == 'udm.Dictionary':
+            data["dictionaries"].append(ci_id)
+        else:
+            data["members"].append(ci_id)
         self.update_ci(env_id, json.dumps(data), 'json')
 
         get_env_response = self.get_ci(env_id, 'xml')
```
Comment (jdewinne) [inline src/main/resources/xlr_xldeploy/XLDeployClient.py]: Kinda weird, but why is this call happening 2x? Once using `json`, once using `xml`. Shouldn't it just be enough to use the `json` code?

**Review by jdewinne — CHANGES_REQUESTED** [review]
Can you also update `remove_ci_from_environment`?

File: src/main/resources/xlr_xldeploy/XLDeployClient.py (hunk 1 above)
Comment (ankurtrivedi) [reply src/main/resources/xlr_xldeploy/XLDeployClient.py]: I tested after removing the json and it does not work. pressed for time so did not go further. Can we add a issue and get to that later?

Comment (jdewinne) [reply src/main/resources/xlr_xldeploy/XLDeployClient.py]: No. Adding more technical debt to something that we can't explain isn't a good practice.
So what do you get when using the `json` only variant?

File: src/main/resources/xlr_xldeploy/XLDeployClient.py [hunk 2]
```diff
@@ -373,26 +373,32 @@ def update_ci_property(self, ci_id, ci_property, property_value):
...
+        else:
+            data["members"].append(ci_id)
         self.update_ci(env_id, json.dumps(data), 'json')
 
-        get_env_response = self.get_ci(env_id, 'xml')
-        items = get_env_response.partition('</members>')
-        xml = items[0] + '<ci ref="' + ci_id + '"/>' + items[1] + items[2]
-        print(xml)
-        self.update_ci(env_id, xml, 'xml')
+        #get_env_response = self.get_ci(env_id, 'xml')
```
Comment (jdewinne) [inline src/main/resources/xlr_xldeploy/XLDeployClient.py]: Can you remove the comments? I trust github we can always find it back.

File: src/main/resources/xlr_xldeploy/XLDeployClient.py [hunk 3]
```diff
@@ -373,26 +373,32 @@ def update_ci_property(self, ci_id, ci_property, property_value):
...
+        #self.update_ci(env_id, xml, 'xml')
 
     def remove_ci_from_environment(self, env_id, ci_id):
         get_env_response = self.get_ci(env_id, 'xml')
         print get_env_response
         env_root = ET.fromstring(get_env_response)
         member_to_remove = None
         for child in env_root:
-            if child.tag == 'members':
+            if child.tag == 'members' or 'dictionaries':
```
Comment (jdewinne) [inline src/main/resources/xlr_xldeploy/XLDeployClient.py]: shouldn't this be `child.tag in ('members', 'dictionaries')`

**Review by jdewinne — CHANGES_REQUESTED** [review]
Can you also run a licenseFormatMain and licenseFormatTest to fix the Travis build

**Pull request merged**